### DISCUSSION TITLE
Force username mapper for Identity Broker authentication

### DISCUSSION THREAD

**Discussion by thannaske** [keycloak/keycloak#33892] [Q&A] [answered]
I'm using a custom GitLab OIDC identity broker for third party authentication. When using a Username Template Importer, the username is pre-populated with `gitlab.j.doe` where `j.doe` is the actual preferred username. That's the way I want it to be, however the user has the theoretical ability to change that username and to divert from that username template.

Is there any way **forcing** the user to use that username template when coming from an identity broker? I don't want them to freely choose their username.

**Comment by danielFesenmeyer** [comment, marked as answer]
Have you tried to change the user permissions on attribute "username" in the "User Profile"? I'd expect that username cannot be changed by users when when there is no edit permission, and it's not even shown when there is no view permission.

**Reply by thannaske** [reply]
YES! That did the trick. Thanks a lot for pointing me in the right direction!
